engine/runtime/src/main/java/org/enso/interpreter/node/callable/argument/ArgumentSorterNode.java

```diff
@@ -95,13 +97,14 @@ private void executeArguments(Object[] arguments, State state) {
    * @return the provided {@code arguments} in the order expected by the cached {@link Function}
    */
   public MappedArguments execute(Function function, State state, Object[] arguments) {
+    var fn = profileFunction.profile(function);
```

Comment:
This makes `fn` compilation constant (if the `function` value is always the same). Only then access to `fn.preparedArguments`, etc. yields compilation constant as well.

Reply:
Should we include this comment in the code then?

IMHO if it's important enough to explain in the PR, it's probably even more important to have that in the code - as the PR will be lost in the depths of the git-log, but the code will remain.

Reply:
Discussion in PRs, git log messages as well as references to hash commits are essential part of project history. The important thing is: be able to locate the right PR for each changed line. Shouldn't that be the way we'd have to explain what `i++` means in the code...

Reply:
Should our documentation be scattered around PRs?

I agree that discussions on which solution to choose and why are rightly placed here and it would be completely unnecessary and unwanted to replicate them in code. But this reads to me like a comment on what this does, something non-trivial that it was worth commenting - so I don't see why we shouldn't include it next to the code it refers to.

Looking at this code I would never have an idea to look thru the blame to find the PR it was introduced in and search its reviews for a comment - I don't imagine this ever happening unless I'd be desperate to understand what happened. I'd probably try to message the author first though, as getting an explanation that way would be far easier.

Having it in a comment would be the most direct way to access it. Of course we do not have to explain `i++` because it is trivial. Here there is some non-trivial observation that you noted, and IMO it would be good to keep track of it. 

It makes it easier for new developers to understand why things are done the way they are in the interpreter if the most important ones are explained in standalone docs or doc comments. Looking through PRs is IMO not a viable way of looking for documentation - in that case we need to pass the knowledge through word-of-mouth, which is a viable means but ideally I don't think this should be how the majority of our know-how is shared.